lib/accesscontrol/permitted_action_on_object_query.rb

```diff
@@ -46,7 +48,8 @@ def can?(action_id, object_type, object_id)
     def grant(action_id, object_type, object_id)
       PermittedActionOnObject.create!(
         id: SecureRandom.uuid,
-        actor: @actor,
+        actor_type: @actors.keys.first,
+        actor_id: @actors.values.first,
```

Comment:
Thinking about how this is going to be used in our application, it now feels a little strange that `grant` and `revoke` would live in this class. Think about it this way:

```rb
# Create query object at the beginning of the request
access = AccessControl::Query.new(User => 4, Group => [5,9,11], Company => 17)

# Now, grant a permission to the current user:
access.grant(actions[:edit], Lesson, 99)

# Now, grant a permission to a specific group:
access.grant(... # what now?
```

I get that it would make more sense to create a new query object with just that group to grant it permissions, but the API doesn't really make that clear. I wonder if `grant` and `revoke` should be class-level rather than on an instantiated object? I think that would make sense given that those two methods don't need to deal with the cache and wouldn't need to be applied until the next request (and therefore, the next instance of the query object).

```rb
AccessControl.grant(user, actions[:edit], Lesson, 99)
AccessControl.revoke(group, actions[:assign], Lesson)
```

That should just require making this method a class method here and in `AccessControl::Query` and requiring the actor to be passed in. What do you think of that idea?

Calling both @rreinhardt9 and @ehourigan to this one.

Reply:
I'm partial to the object methods:  As I started to implement `revoke` it has become clear that we need to support that function for multiple actors (same for `grant`).  If you take a look at #4 - I've added support for `segment_id` which chains off the initializer (and we will need it for both `grant` and `revoke` as well)

Maybe we should rename our top level object? from `Accessly::Query` to `Accessly::[SOMETHING DIFFERENT]`